src/physics/em/KNInteractor.hh

```diff
@@ -0,0 +1,71 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file KNInteractor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Macros.hh"
+#include "base/Types.hh"
+#include "physics/base/Interaction.hh"
+#include "physics/base/ParticleTrackView.hh"
+#include "physics/base/SecondaryAllocatorView.hh"
+#include "physics/base/Secondary.hh"
+#include "physics/base/Units.hh"
+#include "KNInteractorPointers.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Perform Compton scattering, neglecting atomic binding energy.
+ *
+ * This is a model for the discrete Compton inelastic scattering process. Given
+ * an incident gamma, it adds a single secondary (electron) to the secondary
+ * stack and returns an interaction for the change to the incident gamma
+ * direction and energy. No cutoffs are performed for the incident energy or
+ * the exiting gamma energy.
+ *
+ * \note This performs the same sampling routine as in Geant4's
+ *  G4KleinNishinaCompton, as documented in section 6.4.2 of the Geant4 Physics
+ *  Reference (release 10.6).
+ */
+class KNInteractor
```

Comment:
Even-though it is fine now, as we grow the clash likelyhood is high, so I would not use abbreviations
```suggestion
class KleinNishinaComptonInteractor
```

Reply:
Fair enough. The `KleinNishinaModel` vs `KleinNishinaCompton` distinction in Geant4 doesn't make much since to me since they're both models for Compton scattering; the latter just neglects the electron's atomic binding energy. Can you explain the nomenclature? Is there a better name (`KleinNishinaUnboundCompton`) that we could use?

Reply:
@resnegfk @whokion What would be the most "accurate" name here for the model?

Reply:
Quoting from https://geant4.web.cern.ch/node/1621
"... one process (e.g. G4ComptonScattering) and multiple models that can be registered to the process (...). For instance, the models available for the Compton scattering (to be registered to G4ComptonScattering) are: G4KleinNishinaCompton (default)..." I hope this helps.

Reply:
Based on the discussion, I think it's sufficient then to name the class `KleinNishinaInteractor` with the brief class description of "Perform Compton scattering, neglecting atomic binding energy." It sounds like the Geant4 names of Compton vs Model is arbitrary.

Reply:
In my opinion having different "arbitrary" names for the physics algorithm will possibly/eventually lead to confusion.  I.e. if a celeritas entity implements the same physics algorithm that a Geant4 entity that ought to have the same "core" name or if there is a non arbitrary reason to update it should be clearly distinguished from any Geant4 names (i.e. the worse case scenario is if 2 entity in Geant4 and Celeritas have the same "core" name but implement 2 different/distinct physics algorithms ; this would (easily) lead us to think we are comparing apples to apples when we are not).

Reply:
I think that as long as we document the naming correspondence (e.g. the note in the `KleinNishinaInteractor` code) we should be OK. The issue is partly a "trust" one from my perspective -- is `KleinNishinaCompton` the best unambiguous name for it, or is that "just the way Geant4 does it"? I think that by documenting the corresponding Geant4/V/X classes inside each Celeritas class, and keeping a "global" table of the correspondence, we can get around this.

By the way, in the process of creating the [mapping table](https://github.com/celeritas-project/celeritas/wiki/Geant4-correspondence), I found out that GeantV is kind of guilty of the "worst case": its `KleinNishinaComptonModel` corresponds to `G4KleinNishinaCompton`, not `G4KleinNishinaModel`.

I'm going to leave the class name as is; if and when ambiguity starts becoming a problem, we can revisit this.